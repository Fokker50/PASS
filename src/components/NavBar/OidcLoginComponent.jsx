import React, { useState } from 'react';
import { Input, Box, Button, InputLabel } from '@mui/material';
import { styled } from '@mui/system';
import { LoginButton } from '@inrupt/solid-ui-react';

// Utility Imports
import { SOLID_IDENTITY_PROVIDER } from '../../utils';

// Custom Hook Imports
import { useRedirectUrl } from '../../hooks';

const grey = {
  50: '#f6f8fa',
  100: '#eaeef2',
  200: '#d0d7de',
  300: '#afb8c1',
  400: '#8c959f',
  500: '#6e7781',
  600: '#57606a',
  700: '#424a53',
  800: '#32383f',
  900: '#24292f'
};

const StyledInputElement = styled('input')(
  () => `
    width: 240px;
    font-size: 1rem;
    line-height: 1;
    padding: 12px;
    border-radius: 12px;
    color: ${grey[900]};
    background: ${'#fff'};
    border: 1px solid ${grey[200]};

    &:focus {
      border-color: ${grey[700]};
      border-shadow: ${grey[900]};
    }
  
  
    // firefox
    &:focus-visible {
      outline: 0;
    }
  `
);

const OidcLoginComponent = () => {
  const [oidcIssuer, setOidcIssuer] = useState(SOLID_IDENTITY_PROVIDER);
  const redirectUrl = useRedirectUrl();

  return (
    <>
      <Box sx={{ flexGrow: 1 }} />
      <Box sx={{ flexGrow: 1 }} />
      <InputLabel
        sx={{
          marginRight: '16px',
          color: 'white',
          fontSize: '1.2rem',
          fontWeight: 'bold'
        }}
      >
        Pod Server URL:
      </InputLabel>
      <Input
        disableUnderline
        slots={{ input: StyledInputElement }}
        name="Identity Issuer"
        value={oidcIssuer}
        onChange={(e) => setOidcIssuer(e.target.value)}
      />
      <Box sx={{ marginRight: '32px' }} />
      <LoginButton oidcIssuer={oidcIssuer} redirectUrl={redirectUrl}>
        <Button
          variant="contained"
          type="submit"
          color="secondary"
          size="large"
          onClick={() => {
            localStorage.setItem('oidcIssuer', oidcIssuer);
          }}
        >
          Login
        </Button>
      </LoginButton>
    </>
  );
};

export default OidcLoginComponent;
