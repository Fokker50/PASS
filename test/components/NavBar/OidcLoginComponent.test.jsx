import renderer from "react-test-renderer"
import React from 'react';
import { expect, it } from 'vitest';
import OidcLoginComponent from "../../../src/components/NavBar/OidcLoginComponent";

const render = (jsx) => (renderer.create(jsx).toJSON());

it('renders correctly', () => {
  const component = render(<OidcLoginComponent/>)
  expect(component).toMatchSnapshot();
});
