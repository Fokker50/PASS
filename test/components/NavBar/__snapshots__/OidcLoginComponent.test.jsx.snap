// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
[
  <div
    className="MuiBox-root css-i9gxme"
  />,
  <div
    className="MuiBox-root css-i9gxme"
  />,
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1mxi05f-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      Pod Server URL
    </label>
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1pw81iq-MuiInputBase-root-MuiFilledInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="https://solidcommunity.net/"
      />
    </div>
  </div>,
  <div
    className="MuiBox-root css-120gsus"
  />,
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1xnmwiw-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Login
    </button>
  </div>,
]
`;
